### PR TITLE
Feature User Component

### PR DESCRIPTION
**Pull request by b0mung51**

### CODE REVIEW

**Comment by b0mung51** [comment]
I may or may not have changed page.tsx during testing so don't merge any new code on that page.

**Comment by ahmadgaz** [comment]
> I may or may not have changed page.tsx during testing so don't merge any new code on that page.

Make sure to request a reviewer. Also please follow the PR format outlined in the [github workflow](https://github.com/mlatsjsu/mlatsjsu-website/wiki/Github-Workflow).

File: components/molecules/client/follow-btn/component.tsx
Comment (ahmadgaz) [inline components/molecules/client/follow-btn/component.tsx:1]: Also use a placeholder if you can't pull this component from dev bc this is an unrelated file

File: package.json
Comment (ahmadgaz) [inline package.json:1]: how come package files changed? did you do npm update or just npm install?

File: package-lock.json
Comment (ahmadgaz) [inline package-lock.json:1]: unrelated change

File: components/molecules/client/user/component.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  followersCount,
+  bio,
+  avatarUrl,
+}) => {
+  const [isFollowing, setIsFollowing] = useState(initialFollowed);
+
+  const handleFollowToggle = async () => {
+    const action = isFollowing ? 'remove' : 'add';
+    try {
+      const response = await fetch(`/users/${userId}/reading-list`, {
```
Comment (ahmadgaz) [inline components/molecules/client/user/component.tsx:29]: How come this component calling the reading list endpoint? I don't think that exists yet. For now, just handle state, don't make any fetch requests since the backend doesn't have this endpoint yet (afaik).

**Review by ahmadgaz — CHANGES_REQUESTED** [review]
Styling looks great! I requested some changes.

File: package-lock.json
Comment (b0mung51) [inline package-lock.json:1]: Not sure how to revert changes on these since they are "read-only"

File: components/molecules/client/user/component.tsx (hunk 1 above)
Comment (b0mung51) [reply components/molecules/client/user/component.tsx]: I think this is how to handle state, not sure

File: package-lock.json
Comment (ahmadgaz) [reply package-lock.json:1]: Make sure you're on your branch, try `git checkout dev -- package.json`, then commit and push changes

Comment (b0mung51) [reply package-lock.json:1]: Ok reverted the changes to package.json

File: components/molecules/client/user/component.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          <h2 className="text-lg font-semibold">{name}</h2>
+          <p className="text-sm text-gray-500">{followersCount} followers</p>
+          <p className="text-sm text-gray-600">{bio}</p>
+        </div>
+      </div>
+      <FollowBtn
+        variant={isFollowing ? 'unfollow' : 'follow'}
+        //onClick={handleFollowToggle} // Attach the click handler for local state management
+      />
+    </div>
```
Comment (ahmadgaz) [inline components/molecules/client/user/component.tsx:44]: Make sure the styles of the component match the Figma. I went over how to interpret Figma components with Patrick in todays meeting, so check towards the end of the recording or dm me if you have questions.

Also make sure to use colors that are only defined in the tailwind.config.ts file. Any other color won't work unless you are using  brackets [] to use a custom hex or rgb value. If your using vscode make sure you have the tailwindcss extension downloaded.

**Review by ahmadgaz — COMMENTED** [review]
Added some comments, make sure to pull dev to see if the follow-btn works.

Also make sure to export `./user` from the `components/molecules/client/index.ts` file.

File: components/molecules/client/user/component.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  bio,
+  avatarUrl,
+}) => {
+  const [isFollowing, setIsFollowing] = useState(initialFollowed);
+
+  const handleFollowToggle = () => {
+    // Here we just update the local state without making any API requests
+    setIsFollowing(!isFollowing);
+  };
+
```
[the remark this answers is not in the document]
Comment (b0mung51) [reply components/molecules/client/user/component.tsx:29]: done!

File: components/molecules/client/user/component.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        <img src={avatarUrl} alt={name} className="w-12 h-12 rounded-full" />
+        <div>
+          <h2 className="text-lg font-semibold">{name}</h2>
+          <p className="text-sm text-gray-500">{followersCount} followers</p>
+          <p className="text-sm text-gray-600">{bio}</p>
+        </div>
+      </div>
+      <FollowBtn
+        variant={isFollowing ? 'unfollow' : 'follow'}
+        //onClick={handleFollowToggle} // Attach the click handler for local state management
```
[the remark this answers is not in the document]
Comment (b0mung51) [reply components/molecules/client/user/component.tsx:42]: done!

File: components/molecules/client/user/component.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+  const handleFollowToggle = () => {
+    // Here we just update the local state without making any API requests
+    setIsFollowing(!isFollowing);
+  };
+
+  return (
+    <div className="flex items-center justify-between p-4 border-b border-gray-200">
+      <div className="flex items-center space-x-4">
+        <img src={avatarUrl} alt={name} className="w-12 h-12 rounded-full" />
```
[the remark this answers is not in the document]
Comment (b0mung51) [reply components/molecules/client/user/component.tsx:33]: Not sure if I did it correctly

File: components/molecules/client/user/component.tsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
+import FollowBtn from '@/components/molecules/client/follow-btn';
+import Image from 'next/image';
+
+interface UserProps {
+  userId: string;
+  postId: string;
```
Comment (ahmadgaz) [inline components/molecules/client/user/component.tsx]: No need for this?

File: components/molecules/client/user/component.tsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    <div className="flex items-center justify-between w-[460px] h-[89px] gap-md p-4 border border-light-neutral-gray rounded-md">
+      <div className="flex items-center gap-md">
+        {/* Image Box with Fixed Size and Ellipse Shape */}
+        <div className="w-[50px] h-[50px] rounded-full overflow-hidden flex-shrink-0">
+          <Image
+            src={avatarUrl}
+            alt={name}
+            width={50}
+            height={50}
+            className="w-full h-full object-cover"
```
Comment (ahmadgaz) [inline components/molecules/client/user/component.tsx]: add rounded-lg

File: components/molecules/client/user/component.tsx
Comment (ahmadgaz) [inline components/molecules/client/user/component.tsx:1]: One small detail i forgot, there is a title prop that could be passed to the component. It will be optional so you can add it to UserProps like such:
title?: string.

if it's provided, show a little tag next to the name with the title like this:
image

File: components/molecules/client/user/component.tsx [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+          />
+        </div>
+        <div className="flex flex-col justify-center gap-sm w-[329px]">
```
Comment (ahmadgaz) [inline components/molecules/client/user/component.tsx]: Remove "gap-sm" class

File: components/molecules/client/user/user-small.tsx [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+  return (
+    <div className="flex items-center w-[460px] h-[25px] gap-sm">
+      {/* Image Box with Ellipse Shape */}
+      <div className="w-[25px] h-[25px] rounded-full overflow-hidden flex-shrink-0">
+        <Image
+          src={avatarUrl}
+          alt={name}
+          width={25}
+          height={25}
+          className="object-cover w-full h-full"
```
Comment (ahmadgaz) [inline components/molecules/client/user/user-small.tsx]: add "rounded-lg"

File: components/molecules/client/user/component.tsx [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+
+const UserComponent: React.FC<UserProps> = ({
+  userId,
+  name,
+  followersCount,
+  bio,
+  avatarUrl,
+}) => {
+  return (
+    <div className="flex items-center justify-between w-[460px] h-[89px] gap-md p-4 border border-light-neutral-gray rounded-md">
```
Comment (ahmadgaz) [inline components/molecules/client/user/component.tsx]: careful with hard coding w/h. Just use "w-full h-fit"

Comment (ahmadgaz) [reply components/molecules/client/user/component.tsx]: Also for padding, use the spacing values defined in tailwind.config.ts (make sure tailwindcss plugin is installed if you are using vscode)

File: components/molecules/client/user/component.tsx (hunk 8 above)
Comment (ahmadgaz) [reply components/molecules/client/user/component.tsx]: also replace "w-[329px]" with "flex-1". try not to hardcode spacing values. use -xs -sm -md -lg or -xl

File: components/molecules/client/user/component.tsx [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+        <div className="flex flex-col justify-center gap-sm w-[329px]">
+          <h2 className="text-[16px] font-normal leading-[25.6px] text-left text-[#2C353A]">
+            {name}
+          </h2>
+          <p className="text-[13.33px] font-normal leading-[21.33px] text-left text-[#90999E]">
+            {followersCount} followers
+          </p>
+          <p className="text-[13.33px] font-normal leading-[21.33px] text-left text-[#2C353A]">
+            {bio}
+          </p>
```
Comment (ahmadgaz) [inline components/molecules/client/user/component.tsx]: Font styles are defined in tailwind.config.ts. use "text-<style-name>". They are also defined in figma. you can find a texts style name by checking the name under "Typography".  So replace any text styles with whatever font is specified (e.g. text-small-lg).
image

Also colors are defined in tailwind.config.ts (e.g. text-light-neutral-dark)
image

File: components/molecules/client/user/component.tsx [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+        </div>
+      </div>
+      <div className="flex items-start w-[49px] h-[42px]">
```
Comment (ahmadgaz) [inline components/molecules/client/user/component.tsx]: replace w/h values with "w-fit"

File: components/molecules/client/user/user-small.tsx
Comment (ahmadgaz) [inline components/molecules/client/user/user-small.tsx:1]: Replace any hardcoded text styles or spacing values with values from tailwind.config.ts file. Check comments on the component.tsx file to see how

**Review by ahmadgaz — CHANGES_REQUESTED** [review]
Requested some changes

File: components/molecules/client/user/component.tsx (hunk 6 above)
Comment (ahmadgaz) [reply components/molecules/client/user/component.tsx]: Remove the postId property from the interface, you don't need it

File: components/molecules/client/user/component.tsx [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+    <div className="flex items-center justify-between w-full h-full gap-md p-4 border border-light-neutral-gray rounded-md">
+      <div className="flex items-center gap-md">
+        {/* Image Box with Fixed Size and Ellipse Shape */}
+        <div className="w-full h-full rounded-full overflow-hidden flex-shrink-0">
+          <Image
+            src={avatarUrl}
+            alt={name}
+            width={50}
+            height={50}
+            className="w-full h-full object-cover rounded-lg"
```
Comment (ahmadgaz) [inline components/molecules/client/user/component.tsx]: Just for this, leave it as w-[50px] h-[50px]

File: components/molecules/client/user/user-small.tsx [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+  return (
+    <div className="flex items-center w-full h-full gap-sm">
+      {/* Image Box with Ellipse Shape */}
+      <div className="w-fit h-fit rounded-full overflow-hidden flex-shrink-0">
+        <Image
+          src={avatarUrl}
+          alt={name}
+          width={25}
+          height={25}
+          className="object-cover w-full h-full rounded-lg"
```
Comment (ahmadgaz) [inline components/molecules/client/user/user-small.tsx]: Use w-[25px] h-[25px]

File: components/molecules/client/user/component.tsx [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+  name,
+  followersCount,
+  bio,
+  avatarUrl,
+}) => {
+  return (
+    <div className="flex items-center justify-between w-full h-full gap-md p-4 border border-light-neutral-gray rounded-md">
+      <div className="flex items-center gap-md">
+        {/* Image Box with Fixed Size and Ellipse Shape */}
+        <div className="w-full h-full rounded-full overflow-hidden flex-shrink-0">
```
Comment (ahmadgaz) [inline components/molecules/client/user/component.tsx]: don't need to define dimensions here, delete w/h classes

File: components/molecules/client/user/component.tsx [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+
+const UserComponent: React.FC<UserProps> = ({
+  userId,
+  name,
+  followersCount,
+  bio,
+  avatarUrl,
+}) => {
+  return (
+    <div className="flex items-center justify-between w-full h-full gap-md p-4 border border-light-neutral-gray rounded-md">
```
Comment (ahmadgaz) [inline components/molecules/client/user/component.tsx]: remove h-full

**Review by ahmadgaz — CHANGES_REQUESTED** [review]
Just a few more small changes!

File: components/molecules/client/user/component.tsx (hunk 13 above)
Comment (b0mung51) [reply components/molecules/client/user/component.tsx]: Replaced w-full h-full with w-[50px] h-[50px]

File: components/molecules/client/user/user-small.tsx (hunk 14 above)
Comment (b0mung51) [reply components/molecules/client/user/user-small.tsx]: replaced w-full h-full with w-[25px] h-[25px].

File: components/molecules/client/user/component.tsx (hunk 15 above)
Comment (b0mung51) [reply components/molecules/client/user/component.tsx]: Deleted w-full h-full, from div className="w-full h-full rounded-full overflow-hidden flex-shrink-0"

File: components/molecules/client/user/component.tsx (hunk 16 above)
Comment (b0mung51) [reply components/molecules/client/user/component.tsx]: removed h-full

File: components/molecules/client/user/user-small.tsx (hunk 14 above)
Comment (b0mung51) [reply components/molecules/client/user/user-small.tsx]: also removed this for redundancy:
width={25}
          height={25}

File: components/molecules/client/user/component.tsx (hunk 13 above)
Comment (b0mung51) [reply components/molecules/client/user/component.tsx]: Also removed this for redundancy, width={50}
            height={50}

**Review by ahmadgaz — APPROVED** [review]
LGTM
